File: scratch/test_login.js
import mariadb from "mariadb";

const pool = mariadb.createPool({
  host: "gateway01.ap-southeast-1.prod.aws.tidbcloud.com",
  port: 4000,
  user: "Q2KCvyDGKFFGWYC.root",
  password: "Vti3a3KAnPc9C4Cm",
  database: "mediflowai",
  ssl: { rejectUnauthorized: false },
});

async function main() {
  let conn;
  try {
    conn = await pool.getConnection();
    console.log("Successfully connected to DB!");

    const user = await conn.query("SELECT * FROM User WHERE tenantId = 'clinic-290547'");
    console.log("\nParent Tenant User:");
    console.log(user);

    const joined = await conn.query("SELECT su.*, u.subscriptionStatus, u.subscriptionExpiresAt FROM SubUser su JOIN User u ON su.tenantId = u.tenantId WHERE su.email = 'lumsid123@gmail.com'");
    console.log("\nJoined Result:");
    console.log(joined);

  } catch (err) {
    console.error("Error:", err);
  } finally {
    if (conn) conn.release();
    pool.end();
  }
}

main();

File: scratch/test_login.py
import os
import mariadb
import bcrypt

db_host = "gateway01.ap-southeast-1.prod.aws.tidbcloud.com"
db_port = 4000
db_user = "Q2KCvyDGKFFGWYC.root"
db_password = "Vti3a3KAnPc9C4Cm"
db_name = "mediflowai"

try:
    conn = mariadb.connect(
        host=db_host,
        port=db_port,
        user=db_user,
        password=db_password,
        database=db_name,
        ssl_verify_cert=False
    )
    cursor = conn.cursor(dictionary=True)
    print("Successfully connected to DB!")

    # Check users
    cursor.execute("SELECT id, name, email, phone, tenantId FROM User LIMIT 5")
    users = cursor.fetchall()
    print("\nMain Users (Tenants):")
    for u in users:
        print(u)

    # Check sub-users
    cursor.execute("SELECT id, name, email, phone, role, tenantId, isActive FROM SubUser LIMIT 10")
    sub_users = cursor.fetchall()
    print("\nSubUsers:")
    for su in sub_users:
        print(su)

    cursor.close()
    conn.close()
except Exception as e:
    print("Error:", e)
